app/models/proxy.rb

```diff
@@ -114,6 +114,8 @@ class Proxy < ApplicationRecord # rubocop:disable Metrics/ClassLength
   delegate :backend_apis, :backend_api_configs, to: :service
   delegate :scheduled_for_deletion?, to: :account, allow_nil: true
 
+  audited allow_mass_assignment: true
```

Comment:
I assume this is where trouble comes from. It is probably because we are using a custom type `PoliciesConfig` and this somehow confuses audoted. I need to investigate that tomorrow.

Reply:
@RobGri001 , still not sure why this happens in the first place. Seems to be the way rails or psych does serialization or how audited uses that. But there is a workaround that should get you going for the time being.

```diff
--- a/app/models/attributes/policies_config.rb
+++ b/app/models/attributes/policies_config.rb
@@ -13,3 +13,5 @@ class Attributes::PoliciesConfig < ActiveRecord::Type::Text
     new_value != cast(raw_old_value)
   end
 end
+
+PoliciesConfig = Attributes::PoliciesConfig
```

Add this line and it should stop failing. I'll check later whether a better fix will be available.

Reply:
Issue is easy to reproduce and is IMO some bug in `Psych`
```rb
1] pry(main)> pc = Proxy.last.policies_config
=> #<Proxy::PoliciesConfig:0x000000000d074358
 @policies_config=
  [#<Proxy::PolicyConfig:0x000000000d07be50
    @configuration={},
    @enabled=true,
    @name="apicast",
    @version="builtin">,
   #<Proxy::PolicyConfig:0x000000000d07be00
    @configuration={},
    @enabled=true,
    @name="3scale_referrer",
    @version="builtin">]>
[2] pry(main)> ser = pc.to_yaml
=> "--- !ruby/object:PoliciesConfig\npolicies_config:\n- !ruby/object:Proxy::PolicyConfig\n  name: apicast\n  version: builtin\n  configuration: {}\n  enabled: true\n- !ruby/object:Proxy::PolicyConfig\n  name: 3scale_referrer\n  version: builtin\n  configuration: {}\n  enabled: true\n"
[3] pry(main)> YAML.load ser
ArgumentError: undefined class/module PoliciesConfig
from /home/avalon/.asdf/installs/ruby/2.6.7/lib/ruby/2.6.0/psych/class_loader.rb:54:in `path2class'
Caused by ArgumentError: undefined class/module Struct::PoliciesConfig
from /home/avalon/.asdf/installs/ruby/2.6.7/lib/ruby/2.6.0/psych/class_loader.rb:54:in `path2class'
```

This also shows that my above workaround is not ideal. But need more to investigate.

Reply:
I think I see where the issue comes from:

https://github.com/3scale/porta/blob/962e945968721331c5f1d5d9211c95286537067e/app/models/proxy.rb#L565

I have zero idea why this was needed. But it obviously is used by Psych as class name
```ruby
[10] pry(main)> pc.class.name
=> "PoliciesConfig"
```

So one option would be to remove this method. But then something in presentation may break. We can see if any tests would fail. A safer approach might be to extract this class not to be a sub-class of `Proxy` so this method will not be needed anymore as the class will be at the root layer anyway.